crates/chia-sdk-driver/src/layers/nft_ownership_layer.rs

```diff
@@ -1,100 +1,38 @@
 use chia_protocol::Bytes32;
-use chia_puzzles::{
-    nft::{
-        NftOwnershipLayerArgs, NftOwnershipLayerSolution, NftRoyaltyTransferPuzzleArgs,
-        NFT_OWNERSHIP_LAYER_PUZZLE_HASH, NFT_ROYALTY_TRANSFER_PUZZLE_HASH,
-    },
-    singleton::SingletonStruct,
+use chia_puzzles::nft::{
+    NftOwnershipLayerArgs, NftOwnershipLayerSolution, NFT_OWNERSHIP_LAYER_PUZZLE_HASH,
 };
-use chia_sdk_types::conditions::{run_puzzle, NewNftOwner};
-use clvm_traits::{FromClvm, ToNodePtr};
+use clvm_traits::FromClvm;
 use clvm_utils::{CurriedProgram, ToTreeHash, TreeHash};
 use clvmr::{Allocator, NodePtr};
 
-use crate::{DriverError, Puzzle, PuzzleLayer, SpendContext};
+use crate::{DriverError, Layer, Puzzle, SpendContext};
 
 #[derive(Debug)]
-pub struct NftOwnershipLayer<IP> {
+pub struct NftOwnershipLayer<T, I> {
     pub current_owner: Option<Bytes32>,
-    pub launcher_id: Bytes32,
-    pub royalty_puzzle_hash: Bytes32,
-    pub royalty_percentage: u16,
-    pub inner_puzzle: IP,
+    pub transfer_layer: T,
```

Comment:
awesome idea!

Reply:
Yeah I figure we'll probably want to support changing this in the future